lib/fetch/dataURL.js

```diff
@@ -630,14 +630,18 @@ function isASCIIWhitespace (char) {
  * @param {boolean} [trailing=true]
  */
 function removeASCIIWhitespace (str, leading = true, trailing = true) {
-  let i = 0; let j = str.length
+  let lead = 0
+  let trail = str.length - 1
+
   if (leading) {
-    while (j > i && isASCIIWhitespace(str.charCodeAt(i))) --i
```

Comment:
so unless `str.length` was 0 this would always be true... and we're decrementing from i, which means if isASCIIWhitespace(...) was true, `i` would be negative? Not sure why I didn't realize this before.

Reply:
also using `i` and `j` instead of `lead` and `trail` shouldn't have landed